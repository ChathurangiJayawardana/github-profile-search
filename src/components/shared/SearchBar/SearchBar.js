import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import SearchIcon from '../../../assets/images/search-icon.svg';
import colors from '../../../setup/colors';

class SearchBar extends React.Component {
  static propTypes = {
    onSearch: PropTypes.func,
    onChange: PropTypes.func,
  }

  static defaultProps = {
    onSearch: () => null,
    onChange: () => null,
  };

  constructor(props) {
    super(props);
    this.state = { value: '' };
  }

  onChange = (e) => {
    const { value } = e.target;
    const { onChange } = this.props;

    this.setState({ value });
    onChange(value);
  }

  onSearch = () => {
    const { value } = this.state;
    const { onSearch } = this.props;

    onSearch(value);
  }

  render() {
    return (
      <Container>
        <Field
          name="search"
          type="search"
          onChange={this.onChange}
          value={this.state.value}
        />
        <Button onClick={this.onSearch}>
          <img src={SearchIcon} alt="Search Icon" />
        </Button>
      </Container>
    );
  }
}

const Container = styled.div`
  margin-top: 13px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Field = styled.input`
  width: 650px;
  height: 50px;
  padding: 0px 20px;
  font-family: 'Raleway';
  font-size: 20px;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.4);
  font-weight: 300;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  border: none;
`;

const Button = styled.button`
  width: 100px;
  height: 50px;
  border: none;
  cursor: pointer;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.4);
  border-radius: 0px 2px 2px 0px;
  background-color: ${colors.purple}
`;

export default SearchBar;

