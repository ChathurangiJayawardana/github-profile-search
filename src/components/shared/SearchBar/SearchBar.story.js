import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import SearchBar from '.';

const stories = storiesOf('components/SearchBar', module);

const props = {
  style: {},
  value: '',
  onChange: action('onChange Action'),
  onSearch: action('onSearch Action'),
};

stories
  .add('default', () => <SearchBar {...props} />);

