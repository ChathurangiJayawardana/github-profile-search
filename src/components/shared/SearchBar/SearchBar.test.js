import React from 'react';
import { mount, shallow } from 'enzyme';
import { spy } from 'sinon';

import SearchBar from '.';

describe('SearchBar Component', () => {
  describe('renders', () => {
    it('renders without props', () => {
      const wrapper = shallow(<SearchBar />);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with props', () => {
      const rendered = shallow(<SearchBar onChange={jest.fn} onSearch={jest.fn} />);
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('props', () => {
    it('should call onSearch when click on Button', () => {
      const callback = spy();
      const wrapper = mount(<SearchBar onSearch={callback} />);

      wrapper.find('button').simulate('click');

      expect(callback.called).toBe(true);
    })

    it('should call onChange when input change', () => {
      const callback = spy();
      const wrapper = mount(<SearchBar onChange={callback} />);
      const input = wrapper.find('input[name="search"]');

      input.simulate('change', { target: { value: 'keyword', name: 'search' } });

      const value = callback.getCall(0).args[0];

      expect(value).toEqual('keyword');
    });
  });

  describe('content', () => {
    it('should contain a input field', () => {
      const wrapper = mount(<SearchBar onSearch={jest.fn} />);

      expect(wrapper.find('input')).toHaveLength(1);
    });

    it('should contain a button', () => {
      const wrapper = mount(<SearchBar onSearch={jest.fn} />);

      expect(wrapper.find('button')).toHaveLength(1);
    });
  });
});

