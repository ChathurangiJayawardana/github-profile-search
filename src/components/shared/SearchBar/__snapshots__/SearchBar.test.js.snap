// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders renders with props 1`] = `ShallowWrapper {}`;

exports[`SearchBar Component renders renders without props 1`] = `ShallowWrapper {}`;
